spmd/compiler/api.py

```diff
@@ -19,6 +24,7 @@ def __init__(
         optimize_first_iter: bool = False,
         optimizations: Sequence[GraphOptimization] = tuple(),
         map_param_and_grad: bool = True,
+        print_graph: bool = False,
```

Comment:
instead of a flag, consider using logging levels.

log.info(my mostly useful logs)
log.debug(my graph)

then just have users configure the log level

Reply:
This idea is better. Since @anj-s  is preparing another PR for logging the graph. I'll just let she do so and land this PR for now so that we can debug.